RecoTracker/PixelTrackFitting/plugins/PixelTrackDumpAlpaka.cc

```diff
@@ -0,0 +1,80 @@
+#include <Eigen/Core>  // needed here by soa layout
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/RunningAverage.h"
+#include "RecoTracker/TkMSParametrization/interface/PixelRecoUtilities.h"
+
+#include "DataFormats/Vertex/interface/ZVertexSoAHost.h"
+#include "DataFormats/TrackSoA/interface/TrackSoAHost.h"
+
+template <typename TrackerTraits>
+class PixelTrackDumpAlpakaT : public edm::global::EDAnalyzer<> {
+public:
+  using TkSoAHost = TrackSoAHost<TrackerTraits>;
+  using VertexSoAHost = ZVertexHost;
+
+  explicit PixelTrackDumpAlpakaT(const edm::ParameterSet& iConfig);
+  ~PixelTrackDumpAlpakaT() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void analyze(edm::StreamID streamID, edm::Event const& iEvent, const edm::EventSetup& iSetup) const override;
+  edm::EDGetTokenT<TkSoAHost> tokenSoATrack_;
+  edm::EDGetTokenT<VertexSoAHost> tokenSoAVertex_;
+};
+
+template <typename TrackerTraits>
+PixelTrackDumpAlpakaT<TrackerTraits>::PixelTrackDumpAlpakaT(const edm::ParameterSet& iConfig) {
+  tokenSoATrack_ = consumes(iConfig.getParameter<edm::InputTag>("pixelTrackSrc"));
+  tokenSoAVertex_ = consumes(iConfig.getParameter<edm::InputTag>("pixelVertexSrc"));
+}
+
+template <typename TrackerTraits>
+void PixelTrackDumpAlpakaT<TrackerTraits>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("pixelTrackSrc", edm::InputTag("pixelTracksAlpaka"));
+  desc.add<edm::InputTag>("pixelVertexSrc", edm::InputTag("pixelVerticesAlpaka"));
+  descriptions.addWithDefaultLabel(desc);
+}
+
+template <typename TrackerTraits>
+void PixelTrackDumpAlpakaT<TrackerTraits>::analyze(edm::StreamID streamID,
+                                                   edm::Event const& iEvent,
+                                                   const edm::EventSetup& iSetup) const {
+  auto const& tracks = iEvent.get(tokenSoATrack_);
+  assert(tracks.view().quality());
```

Comment:
Is the purpose only assert that these quantities are non-zero? In that case `test/` seems like a better place for the module (and to me the name is also misleading)